cmd/gardenlet/app/migration.go

```diff
@@ -10,20 +10,28 @@ import (
 
 	"github.com/Masterminds/semver/v3"
 	"github.com/go-logr/logr"
+	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/discovery"
 	"k8s.io/client-go/rest"
+	"k8s.io/component-base/version"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/cluster"
 
+	"github.com/gardener/gardener/imagevector"
 	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
 	"github.com/gardener/gardener/pkg/component/extensions/operatingsystemconfig"
 	"github.com/gardener/gardener/pkg/utils/flow"
 	versionutils "github.com/gardener/gardener/pkg/utils/version"
 )
 
 func (g *garden) runMigrations(ctx context.Context, log logr.Logger, _ cluster.Cluster) error {
+	log.Info("Updating required Gardener-Resource-Manager deployments for managed resource compression support")
+	if err := updateGardenerResourceManager(ctx, log, g.mgr.GetClient()); err != nil {
+		return err
+	}
+
```

Comment:
Is this still required? In the pull request description you note that we intend to use a phased approach, i.e. roll out the decompression support with one release and only with an upcoming release add the compression support. If we assume that all clusters are reconciled at least once between gardener version updates this migration code should not be necessary at all.
WDYT?

Reply:
Resolved via offline discussion.

Summary: Failed shoot clusters may not reconcile for some time.